website/templates/project/view_file.mako

```diff
@@ -104,11 +106,12 @@
     ${parent.javascript_bottom()}
         <script type="text/javascript">
           window.contextVars = $.extend(true, {}, window.contextVars, {
-            renderURL: ${"'{}'".format(render_url) if rendered is None else 'undefined'},
+          renderURL: ${"'{}'".format(render_url.replace("'", "\\'")) if rendered is None else 'undefined'},
```

Comment:
Nope, the issue is that this needs to be a javascript string surrounded by `'` and if there happens to be a `'`' in the filename it will break all the javascript on the page so it just needs to be escaped.
May be better as
```
%if rendered is None:
    variable = undefined
%else:
    variable = ${url | js_str}
%endif
```